docs/proposals/11-cluster-health.md

```diff
@@ -0,0 +1,205 @@
+- If necessary, `kube-controller-manager` could be scaled down as described [above](#possible-actions).
+- Some resource creations (say, `pods`) could be throttled/rate-limited.
+- The `machine-controller-manager` could be either scaled down as described [above](#possible-actions) or be enhanced to avoid (or ramp down) deletion of machines from the affected zones (or `machinedeployments`).
+- Gardener can defer any disruptive updates until later.
+
+### Condition Types of future relevance
+
+This is just a sample set of condition types that might be of future relevance. It is not an exhaustive list.
+The suggested possible actions for these condition types are also just samples and not an exhaustive list.
+
+#### UnhealthyInternalNetwork
+
+If there is a way to detect problems in the cluster network that is severe enough to affect any significant portion of the cluster then this could be encoded in a component that updates the conditions with the `UnhealthyInternalNetwork` condition type.
+
+##### Possible Actions
+
+- Alerts can be raised to the administrator of the cluster as well as the operator/support team on duty.
+- Some resource creations (say, `pods`) could be throttled/rate-limited.
+- Gardener can defer any disruptive updates until later.
+
+#### UnhealthyCloudProvider
+
+If there is information about some issues reported against the cloud provider, it could be updated in the conditions with the `UnhealthyCloudProvider` condition type.
+If there is more specific information about the issue then more specific condition types can be defined.
+
+Such conditions could be updated manually by the administrators, operators on duty based on publications of such issues by the cloud providers.
+If there is a way to automatically scrape such information from well-known locations or APIs then this could even be automated.
+
+##### Possible Actions
+
+- Alerts can be raised to the administrator of the cluster as well as the operator/support team on duty.
+- Controllers such as `kube-controller-manager`, `machine-controller-manager` that manage crucial resources such as `nodes` can be either scaled down as describe [above](#possible-actions) or enhanced to change their behaviour to ramp down their active interventions to the cluster.
+- Some resource creations (say, `pods`) could be throttled/rate-limited.
+- Gardener can defer any disruptive updates until later.
+
+#### CloudProviderQuotaExceeded
+
+If there is a way to detect that some cloud provider quotas are exceeded this information can be updated in the conditions with the `CloudProviderQuotaExceeded` condition type.
+If more specific information is available about which particular quota is exceeded then more specific condition types can be defined.
+
+##### Possible Actions
+
+- Alerts can be raised to the administrator of the cluster as well as the operator/support team on duty.
```

Comment:
It does not make sense to send alert to the Gardener operator because usually he can do nothing. Instead, the shoot can be labeled as end-user issue. This will allow filtering out user problems to be filtered out easily from all faulty clusters.

Reply:
> It does not make sense to send alert to the Gardener operator because usually he can do nothing. Instead, the shoot can be labeled as end-user issue. This will allow filtering out user problems easily from all faulty clusters.

Agreed. But an alert to the cluster admin (end-user) might be useful.